### PR TITLE
Fix regression after merge conflict: big font for messages with only big emoji

### PR DESCRIPTION
**Pull request by bmarty**

### CODE REVIEW

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/factory/MessageItemFactory.kt [hunk 1]
```diff
@@ -271,7 +275,8 @@ class MessageItemFactory @Inject constructor(
                 editEnd,
                 Spanned.SPAN_INCLUSIVE_EXCLUSIVE)
 
-        spannable.setSpan(RelativeSizeSpan(.9f), editStart, editEnd, Spanned.SPAN_INCLUSIVE_EXCLUSIVE)
+        // Note: text size is set to 14sp
+        spannable.setSpan(AbsoluteSizeSpan(dimensionConverter.spToPx(13)), editStart, editEnd, Spanned.SPAN_INCLUSIVE_EXCLUSIVE)
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/factory/MessageItemFactory.kt]: Maybe split the line

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/factory/MessageItemFactory.kt [hunk 2]
```diff
@@ -241,12 +243,13 @@ class MessageItemFactory @Inject constructor(
         return MessageTextItem_()
                 .apply {
                     if (informationData.hasBeenEdited) {
-                        val spannable = annotateWithEdited(linkifiedBody, callback, informationData)
+                        val spannable = annotateWithEdited(linkifiedBody, callback, attributes.dimensionConverter, informationData)
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/factory/MessageItemFactory.kt]: Why taking dimension converter from attributes? It should be injected in MessageItemFactory constructor instead, right?

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/item/AbsMessageItem.kt [hunk 3]
```diff
@@ -172,6 +173,7 @@ abstract class AbsMessageItem<H : AbsMessageItem.Holder> : BaseEventItem<H>() {
             val informationData: MessageInformationData,
             val avatarRenderer: AvatarRenderer,
             val colorProvider: ColorProvider,
+            val dimensionConverter: DimensionConverter,
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/item/AbsMessageItem.kt]: See upper comment about dimensionConverter

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/helper/MessageItemAttributesFactory.kt [hunk 4]
```diff
@@ -42,6 +44,7 @@ class MessageItemAttributesFactory @Inject constructor(
                 informationData = informationData,
                 avatarRenderer = avatarRenderer,
                 colorProvider = colorProvider,
+                dimensionConverter = dimensionConverter,
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/helper/MessageItemAttributesFactory.kt]: See upper comment about dimensionConverter

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/factory/MessageItemFactory.kt (hunk 2 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/factory/MessageItemFactory.kt]: yes

**Comment by ganfra** [comment]
Ok for me!

File: vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/factory/MessageItemFactory.kt (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/factory/MessageItemFactory.kt]: done

**Pull request merged**